src/matcha_ml/state/matcha_state.py

```diff
@@ -134,6 +146,79 @@ def state_exists(cls) -> bool:
         """
         return bool(os.path.isfile(cls.matcha_state_path))
 
+    @staticmethod
+    def _parse_terraform_output_resource_name(output_name: str) -> Tuple[str, str, str]:
+        """Build resource output for each Terraform output.
+
+        Format for Terraform output names is:
+        <resource>_<flavor>_<property>
+        where <resource> is a name found in RESOURCE_NAMES
+
+        Args:
+            output_name (str): the name of the Terraform output.
+
+        Returns:
+            Tuple[str, str, str]: the resource output for matcha.state.
+        """
+        resource_type: str
+
+        for key in RESOURCE_NAMES:
+            if key in output_name:
+                resource_type = key
+                break
+
+        if resource_type is None:
+            print_error(
+                "A valid resource type for the output '{output_name}' does not exist."
+            )
+            raise MatchaInputError()
+
+        flavour_and_resource_name = output_name[len(resource_type) + 1 :]
+
+        flavor, resource_name = flavour_and_resource_name.split("_", maxsplit=1)
+        resource_name = resource_name.replace("_", "-")
+        resource_type = resource_type.replace("_", "-")
+
+        return resource_type, flavor, resource_name
+
+    @staticmethod
+    def build_state_from_terraform_output(
```

Comment:
Similar here, why does this need to be a static function? I'd prefer this to be a non-static method, called via an instance of `MatchaStateService`:

```python
def build_state(...) -> self:
```

Note: I've set the return type to `self` because this will presumably change something about the `MatchaStateService` instance, e.g., update it's internal `_state` variable.

Reply:
I think this and `create_state_file` have to be a staticmethod due to the way that someone has updated the `__init__` to raise an exception if the state does not exist. Let's look at an example:

1. User does not have a state file
2. User provisions
3. We try to create a state file, but first we must initialise a `MatchaState` object
4. We initialise a `MatchaState` object
5. In the init we check if the state file exists
6. As per condition (1), the state file does not exist
7. The init raises an exception
8. This exception is caught and reraised
9. The program exits

We have seen that MatchaStateService cannot exist without a `matcha.state` file, therefore we will not be able to call `create` or `build state`, please see "[Chicken or the egg](https://dictionary.cambridge.org/dictionary/english/chicken-and-egg-situation)" for a reference of what situation this will create.

Reply:
To avoid these two functions being static, which is the preferred behaviour, could we update the `__init__` in a way that would prevent this problem?

Instead of raising an error, perhaps it'd be better to call the internal functions if `self.state_exists() == false`?